### PR TITLE
[feat] add http deployment method

### PR DESCRIPTION
**Pull request by liuminjian**
Issue Number: https://github.com/opencurve/curveadm/issues/328

1.增加http daemon的部署方式，host增加protocol和http port字段
image
image

2.在每个host节点上部署http daemon，可以使用ansible playbook部署，或者在k8s环境下用daemonset部署
image

3.部署http daemon后能正常部署curve集群
image

image

image

遗留问题：目前enter指令不支持http方式

### CODE REVIEW

**Comment by liuminjian** [comment]
@caoxianfei1 PTAL

**Comment by caoxianfei1** [comment]
It seems to only support deploy cluster now, but how to implement other  maintenance operations?

File: internal/task/task/task.go [hunk 1]
```diff
@@ -111,16 +113,22 @@ func (t *Task) executePost(ctx *context.Context) {
...
 		client, err := module.NewSSHClient(*t.sshConfig)
 		if err != nil {
 			return errno.ERR_SSH_CONNECT_FAILED.E(err)
 		}
-		sshClient = client
+		remoteClient = client
+	} else if t.httpConfig != nil {
+		client, err := module.NewHttpClient(*t.httpConfig)
+		if err != nil {
+			return errno.ERR_SSH_CONNECT_FAILED.E(err)
```
Comment (caoxianfei1) [inline internal/task/task/task.go]: ERR_HTTP...

**Comment by caoxianfei1** [comment]
LGTM!

File: internal/task/task/task.go (hunk 1 above)
Comment (liuminjian) [reply internal/task/task/task.go]: ok

**Comment by aspirer** [comment]
the branch you pushed should be `develop`

**Pull request closed without merging**